src/URI.js

```diff
@@ -487,6 +487,36 @@
 
   URI.encodeReserved = generateAccessor('reserved', 'encode');
 
+  URI.toUri = function(string, defaults) {
+    /**
+     * Convert a string which may or may not have a valid scheme
+     * to a valid URI prepending a scheme if needed
+     * @param {string} string   The url to parse
+     * @param {object} defaults The defaults to use for the URI properties if not parsed.
+     *        Currently only supports, defaults.scheme which should be one of [https, http]
+     *        '//' will be added if no scheme is parsed or provided
+     * @return {string} A valid URI
+     *
+     */
+    defaults = defaults || {};
+    // if the uri doesn't have a scheme, add one
+    // now so it doesn't get parsed as a path
+    if (!string.match(/^(https|http)?:\/\//i)) {
```

Comment:
This would also match URLs of the form `ftp://…` which would then produce `http://ftp://…`.

You'd be looking for the index of `"://"` and taking the the substring that comes before. You'd add the `default.scheme` only if `"://"` is not part of the string, or the substring before that is the empty string.

https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L218 is a ready-to-use expression for validating a scheme, if should that be necessary.